pkg/registry/client.go

```diff
@@ -209,60 +201,127 @@ type (
 	LoginOption func(*loginOperation)
 
 	loginOperation struct {
-		username string
```

Comment:
I think you'd be better off keeping all these values in `loginOperation` as a context and doing all the validation after all the callbacks have been processed. This would avoid the error in member and the `panic` calls. 

I don't like this options/operations pattern in general, but that is another matter.